src/cdef.rs

```diff
@@ -420,9 +421,13 @@ pub fn cdef_padded_frame_copy<T: Pixel>(in_frame: &Frame<T>) -> Frame<u16> {
     },
   };
 
-  for p in 0..3 {
+  for p in 0..MAX_PLANES {
     let rec_w = in_frame.planes[p].cfg.width;
     let rec_h = in_frame.planes[p].cfg.height;
+
+    /* Its a monochrome frame but we have no way to signal that */
+    if rec_w == 0 || rec_h == 0 { break; };
```

Comment:
We can leave this out for some small overhead when encoding monochrome, but if possible I'd rather have a better solution.

Reply:
This optimization is definitely worth keeping. Passing in # of planes or putting it in the Frame struct are other valid solutions. In Frame might be a bit awkward because it's really an encoder wide setting.